### PR TITLE
Release Candidate 0.3.4

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Release Candidate for pysatMissions.

Adds support for skyfield propagation, and implements a number of maintenance updates to keep in line with pysat and other packages.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

via pytest

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zeondo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs: https://pysatmissions.readthedocs.io/en/rc_0_3_4/
- [x] Upload the test distribution to the testPyPi server: https://test.pypi.org/project/pysatMissions/0.3.4rc0/
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages) - na
- [x] Create a wiki page with the release text: https://github.com/pysat/pysatMissions/wiki/v0.3.4
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -8,25 +8,22 @@ on: [push, pull_request]
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ["ubuntu-latest"]
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.9]
+        python-version: ["3.9"]
```
Comment (aburrell) [inline .github/workflows/docs.yml:15]:
```suggestion
        python-version: ["3.11"]
```

File: .github/workflows/pip_rc_install.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10"]  # Keep this version at the highest supported Python version
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml:15]:
```suggestion
        python-version: ["3.11"]  # Keep this version at the highest supported Python version
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  "Topic :: Scientific/Engineering :: Astronomy",
+  "Topic :: Scientific/Engineering :: Physics",
+  "Topic :: Scientific/Engineering :: Atmospheric Science",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: BSD License",
+  "Natural Language :: English",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
```
Comment (aburrell) [inline pyproject.toml:26]:
```suggestion
  "Programming Language :: Python :: 3.9",
  "Programming Language :: Python :: 3.10",
  "Programming Language :: Python :: 3.11",
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,88 @@
+[build-system]
+requires = ["setuptools", "pip >= 10"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pysatMissions"
+version = "0.3.4"
+description = "Mission Planning toolkit for pysat"
+readme = "README.md"
+requires-python = ">=3.8"
```
Comment (aburrell) [inline pyproject.toml:10]:
```suggestion
requires-python = ">=3.9"
```

File: pysatMissions/tests/test_methods_spacecraft.py [hunk 5]
```diff
@@ -73,29 +79,61 @@ def test_calculate_ecef_velocity(self):
...
+
+        # Xhat vector should match velocity
+        assert np.all(self.testInst['sc_xhat_ecef_x']
+                      == self.testInst['velocity_ecef_x'])
+        assert np.all(self.testInst['sc_xhat_ecef_y']
+                      == self.testInst['velocity_ecef_y'])
+        assert np.all(self.testInst['sc_xhat_ecef_z']
+                      == self.testInst['velocity_ecef_z'])
+
+        # Zhat vector should match - position
```
Comment (aburrell) [inline pysatMissions/tests/test_methods_spacecraft.py:109]:
```suggestion
        # Z-hat vector should match - position
```

File: pysatMissions/tests/test_methods_spacecraft.py [hunk 6]
```diff
@@ -73,29 +79,61 @@ def test_calculate_ecef_velocity(self):
...
+        assert np.all(self.testInst['sc_zhat_ecef_x']
+                      == -self.testInst['position_ecef_x'])
+        assert np.all(self.testInst['sc_zhat_ecef_y']
+                      == -self.testInst['position_ecef_y'])
+        assert np.all(self.testInst['sc_zhat_ecef_z']
+                      == -self.testInst['position_ecef_z'])
+
+        # Yhat vector should be orthogonal
```
Comment (aburrell) [inline pysatMissions/tests/test_methods_spacecraft.py:117]:
```suggestion
        # Y-hat vector should be orthogonal
```

**Review by aburrell — APPROVED** [review]

**Comment by jklenzing** [comment]
https://github.com/orgs/pysat/projects/6?pane=issue&itemId=12664459

File: pysatMissions/tests/test_instruments.py [hunk 7]
```diff
@@ -33,6 +33,8 @@
 for inst in instruments['download']:
     if 'sgp4' in inst['inst_module'].name:
         instruments['sgp4'].append(inst)
+    if 'skyfield' in inst['inst_module'].name:
+        instruments['sgp4'].append(inst)
```
Comment (JonathonMSmith) [inline pysatMissions/tests/test_instruments.py:37]: This means that skyfield instruments are added to the list of sgp4 instruments right?

Comment (jklenzing) [reply pysatMissions/tests/test_instruments.py:37]: Yes. This is for tests that are not run on ephem

**Review by JonathonMSmith — APPROVED** [review]

**Pull request merged**